mongodbatlas/resource_mongodbatlas_federated_database_instance.go

```diff
@@ -47,31 +49,40 @@ func resourceMongoDBAtlasFederatedDatabaseInstance() *schema.Resource {
 					Type: schema.TypeString,
 				},
 			},
-			"aws": {
+			"cloud_provider_config": {
 				Type:     schema.TypeList,
 				MaxItems: 1,
 				Optional: true,
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
-						"role_id": {
-							Type:     schema.TypeString,
-							Required: true,
-						},
-						"test_s3_bucket": {
-							Type:     schema.TypeString,
+						"aws": {
+							Type:     schema.TypeList,
+							MaxItems: 1,
 							Required: true,
-						},
-						"iam_assumed_role_arn": {
-							Type:     schema.TypeString,
-							Computed: true,
-						},
-						"iam_user_arn": {
-							Type:     schema.TypeString,
-							Computed: true,
-						},
-						"external_id": {
-							Type:     schema.TypeString,
-							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"role_id": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"test_s3_bucket": {
+										Type:     schema.TypeString,
+										Required: true,
```

Comment:
Is s3 required on the cloud provider config today or is the value optional? Wondering how the provider is currently workign without the field.

Reply:
`test_s3_bucket` is required inside `aws`. `cloud_provider_config` is optional